### ISSUE TITLE
PSR12.ControlStructures.ControlStructureSpacing not ignoring indentation inside multi-line string arguments

### ISSUE THREAD

**Issue by dereuromark** [squizlabs/PHP_CodeSniffer#2623]
```php
if (preg_match(
            '(
                \*\s+[{}A-Z0-9\-]+.*\s+
                \*?\s*
                \*\s+@api
            )xi',
            $comment
        )) {
            return;
        }
```
reports
```
 86 | ERROR | [x] Each line in a multi-line control structure must be indented at least once; expected at least 12 spaces, but found 0
    |       |     (PSR12.ControlStructures.ControlStructureSpacing.LineIndent)
 88 | ERROR | [x] Each line in a multi-line control structure must be indented at least once; expected at least 12 spaces, but found 8
    |       |     (PSR12.ControlStructures.ControlStructureSpacing.LineIndent)
 88 | ERROR | [x] The closing parenthesis of a multi-line control structure must be on the line after the last expression
    |       |     (PSR12.ControlStructures.ControlStructureSpacing.CloseParenthesisLine)
```
but when running it

> FAILED TO FIX

Using -vvv it seems that this is what it fixes it too, breaking other rules:
```
if (preg_match(
 82|            '(
 83|                \*\s+[{}A-Z0-9\-]+.*\s+
 84|                \*?\s*
 85|                \*\s+@api
 86|                                                                                                                                                                                                )xi',
 87|            $comment
 88|            )
 89|        ) {
 90|            return;
 91|        }
```

**Comment by spaceemotion** [comment]
Not sure if this is related, but the following code gets completely broken in terms of layout:

```php
return trim(preg_replace_callback(
            // sprintf replaces IGNORED_CHARS multiple times: for %s as well as %1$s (argument numbering)
            // /[%s]*([^%1$s]+)/ results in /[IGNORED_CHARS]*([^IGNORED_CHARS]+)/
                sprintf('/[%s]*([^%1$s]+)/', self::IGNORED_CHARS),
                function (array $term) use ($mode): string {
                    // query pieces have to bigger than one char, otherwise they are too expensive for the search
                    if (mb_strlen($term[1], 'UTF-8') > 1) {
                        // in boolean search mode '' (empty) means OR, '-' means NOT
                        return sprintf('%s%s ', $mode === 'AND' ? '+' : '', self::extractUmlauts($term[1]));
                    }

                    return '';
                },
                $search
            ));
```

with the psr-12 preset becomes:

```php
return trim(preg_replace_callback(
    // sprintf replaces IGNORED_CHARS multiple times: for %s as well as %1$s (argument numbering)
            // /[%s]*([^%1$s]+)/ results in /[IGNORED_CHARS]*([^IGNORED_CHARS]+)/
                sprintf('/[%s]*([^%1$s]+)/', self::IGNORED_CHARS),
    function (array $term) use ($mode): string {
                    // query pieces have to bigger than one char, otherwise they are too expensive for the search
        if (mb_strlen($term[1], 'UTF-8') > 1) {
            // in boolean search mode '' (empty) means OR, '-' means NOT
            return sprintf('%s%s ', $mode === 'AND' ? '+' : '', self::extractUmlauts($term[1]));
        }

                    return '';
    },
    $search
));
```

expected:

```php
return trim(preg_replace_callback(
    // sprintf replaces IGNORED_CHARS multiple times: for %s as well as %1$s (argument numbering)
    // /[%s]*([^%1$s]+)/ results in /[IGNORED_CHARS]*([^IGNORED_CHARS]+)/
    sprintf('/[%s]*([^%1$s]+)/', self::IGNORED_CHARS),
    function (array $term) use ($mode): string {
        // query pieces have to bigger than one char, otherwise they are too expensive for the search
        if (mb_strlen($term[1], 'UTF-8') > 1) {
            // in boolean search mode '' (empty) means OR, '-' means NOT
            return sprintf('%s%s ', $mode === 'AND' ? '+' : '', self::extractUmlauts($term[1]));
        }

        return '';
    },
    $search
));
```

**Comment by gsherwood** [comment]
> Not sure if this is related (if not, i can open up a new issue), but the following code gets completely broken in terms of layout:

Different issue.

**Comment by gsherwood** [comment]
Thanks for the report. The sniff was not ignoring multi-line string arguments and so was complaining about indentation and conflicting with other rules.

The sample code is now fixed like this:
```php
<?php
if (
    preg_match(
        '(
                \*\s+[{}A-Z0-9\-]+.*\s+
                \*?\s*
                \*\s+@api
            )xi',
        $comment
    )
) {
            return;
}
```
